src/bun.js/api/server.zig

```diff
@@ -198,6 +198,37 @@ pub const AnyRoute = union(enum) {
             }
         }
 
+        if (argument.as(jsc.WebCore.Response)) |response| {
+            if (response.body.value == .HTMLBundle) {
+                const html_bundle = response.body.value.HTMLBundle;
+                const needs_custom = response.init.headers != null or response.statusCode() != 200;
+
+                if (needs_custom) {
```

Comment:
going to rebundle just for the headers?

Reply:
It was simplest way to add to context. I couldn't use the existing cache as otherwise only one config option would have been used even if user set many diferent ones.

Ideally they should be seperate and independent things though.